google-cloud-nio/src/main/java/com/google/cloud/storage/contrib/nio/CloudStorageFileSystemProvider.java

```diff
@@ -987,7 +987,7 @@ public boolean requesterPays(String bucketName) {
    * requester-pays.
    */
   public CloudStorageFileSystemProvider withNoUserProject() {
-    return new CloudStorageFileSystemProvider("", this.storageOptions);
+    return new CloudStorageFileSystemProvider(null, this.storageOptions);
```

Comment:
This matches how I've seen providers created when there's no project specified.

Reply:
To me this does not seem like a necessary change if we are already modifying the check in CloudStoragePath.java to check for both empty and null. I think we should either change this everywhere or change it nowhere and I think a refactoring change to make it consistent everywhere can be done at a later date.

Reply:
It's not necessary, I can remove it.  It just seemed weird that if you specify no user project then you get a null, but when it removes the user project you get an empty string.  I think a refactoring pass to make it consistent would good in order to prevent similar bugs in the future though.